src/signature.c

```diff
@@ -157,6 +157,118 @@ GBytes *cms_sign(GBytes *content, const gchar *certfile, const gchar *keyfile, G
 	return res;
 }
 
+gchar* print_signer_cert(STACK_OF(X509) *verified_chain) {
+	BIO *mem;
+	gchar *data;
+
+	g_return_val_if_fail(verified_chain != NULL, NULL);
+
+	mem = BIO_new(BIO_s_mem());
+	X509_print_ex(mem, sk_X509_value(verified_chain, 0), 0, 0);
+
+	BIO_get_mem_data(mem, &data);
+
+	BIO_set_close(mem, BIO_NOCLOSE); /* So BIO_free() leaves BUF_MEM alone */
+	BIO_free(mem);
+
+	return data;
+}
+
+gchar* print_cert_chain(STACK_OF(X509) *verified_chain) {
+	GString *text = g_string_new(NULL);
+	char buf[BUFSIZ];
+
+	g_string_append(text, "Certificate Chain:\n");
+	for (int i = 0; i < sk_X509_num(verified_chain); i++) {
+		X509_NAME_oneline(X509_get_subject_name(sk_X509_value(verified_chain, i)),
+				buf, sizeof buf);
+		g_string_append_printf(text, "%2d s:%s\n", i, buf);
```

Comment:
use "Subject" instead of "s"

Reply:
Ok, got that change in my follow-up PR #155 but I agree we can do it here already